src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,117 @@
+#include <aikido/constraint/SequentialSampleable.hpp>
+
+#include <dart/common/StlHelpers.hpp>
+
+namespace aikido {
+namespace constraint {
+
+// TODO (avk): Should I declare the class first and then define?
+// What is "good" C++ convention? Don't tell me it's a preference JS! -.-
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>> generators)
+    : mStateSpace(std::move(stateSpace))
+    , mGenerators(std::move(generators))
+    , mIndex(0)
+  {
+    // Do nothing
+    // TODO (avk): Should we again check equivalence of statespaces between
+    // that of each generator and mStateSpace? I don't think it's necessary.
```

Comment:
Let's use assertion in this case; we're pretty sure this case never happen but just ensure that in debug mode.